src/internal/operators/buckets.ts

```diff
@@ -0,0 +1,135 @@
+import { Observable } from '../Observable';
+import { Subject } from '../Subject';
+import { OperatorFunction } from '../types';
+import { operate } from '../util/lift';
+import { OperatorSubscriber } from './OperatorSubscriber';
+
+export interface BucketsOptions<T> {
+  /**
+   * The hash function used to partition the values into their buckets. The
+   * default hash function converts the values into numbers, if they are not
+   * already.
+   */
+  hashFn?: (value: T) => number;
```

Comment:
Personally, I'm not a fan of abbreviations. I'd prefer `hasher`, instead, but this can be bikeshedded later.

Reply:
i also just picked this up from somewhere, bikeshed away ;)